gradle-jdks-setup/src/main/java/com/palantir/gradle/jdks/setup/CaResources.java

```diff
@@ -53,113 +62,73 @@ public CaResources(ILogger logger) {
-                break;
-        }
-    }
-
-    private void unixImportCertInJdk(AliasContentCert aliasContentCert, Path jdkInstallationDirectory) {
+    private void importCertificates(Path jdkInstallationDirectory, List<Certificate> certificates) {
         try {
-            if (isCertificateInTruststore(jdkInstallationDirectory, aliasContentCert.getAlias())) {
-                logger.log("Certificate already exists in the truststore, skipping...");
-                return;
+            char[] passwd = "changeit".toCharArray();
+            Path jksPath = jdkInstallationDirectory.resolve("lib/security/cacerts");
+            KeyStore jks =
+                    loadKeystore(passwd, jksPath).orElseThrow(() -> new RuntimeException("Could not load keystore"));
+
+            for (Certificate cert : certificates) {
+                String alias = getAlias((X509Certificate) cert);
+                logger.log(String.format("Certificate %s imported successfully into the KeyStore.", alias));
+                jks.setCertificateEntry(alias, cert);
             }
-            File palantirCertFile = File.createTempFile(aliasContentCert.getAlias(), ".pem");
-            Files.write(palantirCertFile.toPath(), aliasContentCert.getContent().getBytes(StandardCharsets.UTF_8));
-            String keytoolPath = jdkInstallationDirectory
-                    .resolve("bin/keytool")
-                    .toAbsolutePath()
-                    .toString();
-            List<String> importCertificateCommand = List.of(
-                    keytoolPath,
-                    "-import",
-                    "-trustcacerts",
-                    "-alias",
-                    aliasContentCert.getAlias(),
-                    "-cacerts",
-                    "-storepass",
-                    "changeit",
-                    "-noprompt",
-                    "-file",
-                    palantirCertFile.getAbsolutePath());
-            CommandRunner.runWithLogger(
-                    new ProcessBuilder(importCertificateCommand), this::writeStdOutput, this::writeStdError);
-        } catch (IOException e) {
-            throw new RuntimeException("Unable to import the certificate to the jdk", e);
-        }
-    }
 
-    private void writeStdOutput(InputStream inputStream) {
-        CommandRunner.processStream(inputStream, logger::log);
-    }
+            jks.store(new BufferedOutputStream(new FileOutputStream(jksPath.toFile())), passwd);
 
-    private void writeStdError(InputStream inputStream) {
-        CommandRunner.processStream(inputStream, logger::logError);
+            logger.log("Certificates imported successfully into the KeyStore.");
+        } catch (KeyStoreException | FileNotFoundException e) {
+            throw new RuntimeException("Failed to import certificates", e);
+        } catch (CertificateException | IOException | NoSuchAlgorithmException e) {
+            throw new RuntimeException(e);
```

Comment:
I'd just unify all these catches into one with the "Failed to import certificates" message